cachito/workers/pkg_managers/gomod.py

```diff
@@ -723,37 +727,49 @@ def _get_highest_semver_tag(repo, target_commit, major_version, all_reachable=Fa
     not_semver_tag_msg = "%s is not a semantic version tag"
     highest = None
     for tag_name in tag_names:
-        if not tag_name.startswith("v"):
-            log.debug(not_semver_tag_msg, tag_name)
-            continue
-
         try:
-            # Exclude the 'v' prefix since this is required by Go, but it is seen as invalid by
-            # the semver Python package
-            parsed_version = semver.parse_version_info(tag_name[1:])
+            semantic_version = _get_semantic_version_from_tag(tag_name, subpath)
         except ValueError:
             log.debug(not_semver_tag_msg, tag_name)
             continue
```

Comment:
Where are we filtering the tags that match the subpath? Is it here or did I miss something else?

Reply:
Well, the `_get_semantic_version_from_tag` will try to remove the `subpath` from a tag, so in case we have:

`sub1/v0.1.1`
`sub2/v0.1.1`

And we're processin `sub1`, `sub2/v0.1.1` will not be properly parsed and will be ignored. 

But now I'm thinking... if the root module has semver tags, it will also fail because the script will not replace the `v`, although this behavior is not obvious.

I will add some more unit test cases to have this covered... might refactor things a bit.

Reply:
+1, also if there's an incorrect root tag (like `1.0.1`, missing the `v`) and you're processing `submodule/v1.0.0`, I think the root tag would get parsed successfully (and be incorrectly applied to the submodule)

Reply:
Great catch! That was indeed a bug. I changed the filtering logic so it is more explicit now.